packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -331,9 +334,22 @@ class _ProductListPageState extends State<ProductListPage>
                     : appLocalizations.product_could_not_remove,
               ),
               duration: SnackBarDuration.medium,
+              action: SnackBarAction(
+                textColor: PRIMARY_BLUE_COLOR,
+                label: removed ? appLocalizations.undo : '',
+                onPressed: () async {
+                  if (!removed) {
+                    return;
+                  }
+                  barcodes.insert(index, barcode);
+                  _selectedBarcodes.add(barcode);
```

Comment:
why are we putting it back here in  `_selectedBarcodes` , we aren't sure if it was there or not, 
I think what would make sense is checking if it was there in the `_selectedBarcodes` then only put it back

Reply:
Because it was removed when we deleted the card ! Even it's in selected barcodes before after deleting is it necessary to put that card in selected barcodes ? If not will remove that !

Reply:
I think when we called the method to remove it doesn't necessarily mean it did remove it, 

The remove function works like 
If it's present there remove it and return true, or return false if there's isn't this element to remove.
[Ref](https://api.flutter.dev/flutter/dart-core/List/remove.html)

Reply:
yeah !!! got it. I will remove that line!

Reply:
I think good for now, Just a  minor comment

You could have checked if the barcode was removed from the selected_barcodes, on line  322
and based on that variable, u could reinsert the barcode in the selected_barcodes, when u undoing it,

Like if it was removed then add back or ignore